python/ray/util/sgd/torch/torch_runner.py

```diff
@@ -193,9 +199,18 @@ def train_epoch(self, num_steps=None, profile=False, info=None):
             SCHEDULER_STEP: self.scheduler_step_freq
         })
         with self.timers.record("train_epoch"):
-            iterator = self.train_loader
+            if iterator is None:
+                iterator = iter(self.train_loader)
+            else:
+                # Dataset will provide us with a list of tuples but we
+                # need two lists.
+                def format_batch(batch):
+                    features, targets = zip(*batch)
+                    return torch.cat(features), torch.cat(targets)
+
+                iterator = map(format_batch, iterator)
             if num_steps:
-                iterator = itertools.islice(iter(self.train_loader), num_steps)
+                iterator = itertools.islice(iterator, num_steps)
             train_stats = self.training_operator.train_epoch(iterator, info)
```

Comment:
@maximsmol what did you intend to do here?

Reply:
Note that line 203 calls iter on the train_loader.

Reply:
@richardliaw are you talking about my "downstream state" API proposal?